File: client/src/app/core/types/portfolioInfoDto.ts
import {PortfolioValuesByDates} from './portfolio-values-by-dates';

export interface PortfolioInfoDto {
    valueIncreasePct: number;
    currencyCode: string;
    investedValueByDates: PortfolioValuesByDates;
    marketValueByDates: PortfolioValuesByDates;
}

File: src/main/java/com/topably/assets/portfolios/domain/dto/pub/PortfolioInfoDto.java
package com.topably.assets.portfolios.domain.dto.pub;

import com.topably.assets.portfolios.domain.dto.PortfolioValuesByDates;
import lombok.Data;
import lombok.experimental.Accessors;

import java.math.BigDecimal;

@Data
@Accessors(chain = true)
public class PortfolioInfoDto {

    private BigDecimal valueIncreasePct;
    private String currencyCode;
    private PortfolioValuesByDates investedValueByDates;
    private PortfolioValuesByDates marketValueByDates;
}
